### ISSUE TITLE
build_setup.sh: export: `BAZEL_BUILD_OPTIONS+': not a valid identifier

### ISSUE THREAD

**Issue by krissell** [envoyproxy/envoy#26921]
*Title*: v1.26.0 - build_setup.sh: line 125: export: `BAZEL_BUILD_OPTIONS+': not a valid identifier

*Description*:
When running build_setup.sh against v1.26.0 I got the above error. Looking at the main branch, this is now line 145.

It looks like the export is happening before variable append.

**Comment by phlax** [comment]
trying to understand the issue here - this file is sourced by the bazel ci runner

with the caveat that i just landed a bunch of cleanups (and ~more/less related fixes) i think it works as expected

generally i export and set separately but in this case have not - either way i think this is valid bash and the var is set just above

im wondering how you are running this and whether it may be related to your bash env

**Comment by krissell** [comment]
Hi @phlax,  I'm calling ./ci/run_envoy_docker.sh, passing it ./ci/do_ci.sh, which then calls ./ci/build_setup.sh

However I was able to reproduce the error with a simple bash script:
      #!/bin/bash
      BAZEL_BUILD_OPTIONS=startValue
      export BAZEL_BUILD_OPTIONS+=("--test_tmpdir=/tmp}")
      echo ${BAZEL_BUILD_OPTIONS}

Calling it:
./test.sh
./test.sh: line 3: export: `BAZEL_BUILD_OPTIONS+': not a valid identifier

**Comment by phlax** [comment]
i think there is some issue with your env - the script is not correct - but bash should should be tolerant to that

a more correct bash script with an array

```bash

#!/bin/bash
BAZEL_BUILD_OPTIONS=()
# no point exporting an array
BAZEL_BUILD_OPTIONS+=("--test_tmpdir=/tmp}")
echo ${BAZEL_BUILD_OPTIONS[@]}

```

**Comment by krissell** [comment]
Thanks for the fast reply @phlax . I'll crawl through out environment to see if I can find what could be causing it.  If you happen to be in the code though, and could make it 'more correct' as you have above, I'd appreciate it.

**Comment by phlax** [comment]
if there is a specific change that you think is a good idea, please PR, i believe the code is ~correct, but its being actively worked on

**Comment by krissell** [comment]
I submitted PR https://github.com/envoyproxy/envoy/pull/26941/files
I've not contributed to the project yet, but this seems so trivial enough

**Comment by adisuissa** [comment]
Fixed by #26941, closing.
